File: Java/dependency-injection/control-freak-refactored/src/main/java/com/github/grzesiek_galezowski/DependencyInjectionAfter/Outbound/XmlOutboundMessageFactory.java
package com.github.grzesiek_galezowski.DependencyInjectionAfter.Outbound;

public class XmlOutboundMessageFactory implements IOutboundMessageFactory {
  public IOutboundMessage CreateOutboundMessage() {
    return new XmlOutboundMessage(new XmlMarshalling());
  }
}

File: CSharp/DependencyInjection/DependencyInjection/DependencyInjectionAfter/_1_ControlFreakRefactoredToDependencyInjection/Outbound/XmlOutboundMessageFactory.cs
namespace DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Outbound
{
  public interface IOutboundMessageFactory
  {
    XmlOutboundMessage CreateOutboundMessage();
  }

  public class XmlOutboundMessageFactory : IOutboundMessageFactory
  {
    public XmlOutboundMessage CreateOutboundMessage()
    {
      return new XmlOutboundMessage(new XmlMarshalling());
    }
  }
}